### ISSUE TITLE
Booking/Order ID's & Invoice Numbers

### ISSUE THREAD

**Issue by cbl** [aw-studio/guidelines#16]
For bookings, orders or invoices, a number/ID format should be agreed upon. This ID should be stored in an extra attribute, e.g.: `#0001`, `a0001`, `b0001`.

**Comment by jannescb** [comment]
super-nice!

**Comment by jannescb** [comment]
```php
"-0000000".sprintf('%07d', $invoicesInSequence + 1);
```

the zeros are unnecessary, right?

**Comment by cbl** [comment]
@jannescb correct. This thread has some usefull tips from others as well. `MAX` is an even better solution than count, since the number gets incremented, even when rows are deleted. At first I thought this wouldn't work since all rows within the sequence need to be locker in order to make sure no number occurs more than once, I will check if `MAX` works like `COUNT` in that case.
